### ISSUE TITLE
[Bug] Incorrect Expected Output Shape (Softmax Attention)

### ISSUE THREAD

**Issue by DhruvSrikanth** [AlphaGPU/leetgpu-challenges#68]
### Problem
- Softmax Attention
- PyTorch variant

### Issue
The shape of `output: Tensor` is of the shape `(M * d)` instead of `(M, d)`. I understand why this is the case for the CUDA variant of the problem but for PyTorch, I'd expect the output shape to reflect the actual output of the attention mechanism.

When returning a tensor of shape `(M, d)`:
<img width="925" height="996" alt="Image" src="https://github.com/user-attachments/assets/51fc18d7-2f52-4ed0-8b1d-4475e51314f5" />

When returning a tensor of shape `(M * d)`:
<img width="911" height="1012" alt="Image" src="https://github.com/user-attachments/assets/7b8abe76-f4b3-4d99-8954-7100fe8c811b" />
